### PR TITLE
Feature fieldmapping readme (PSv2)

### PR DESCRIPTION
**Pull request by rhouthuijzen**
Added fieldmapping
Updated readme to new template
Fixed issue (this is also solved in topdesk operator)

### CODE REVIEW

**Comment by rschouten97** [comment]
@rhouthuijzen the code appears to be well-structured. However, since we are focusing on the PS v2 template, please ensure that the code aligns with the template standards.
I recommend waiting at least a week before making any significant changes, as the template is still undergoing revisions following its recent release.
Furthermore, I recommend merging this pull request only once we have actually implemented this connector.

**Comment by rhouthuijzen** [comment]
Added necessary compare object. Tested on test portal and at two customers

File: update.ps1 [hunk 1]
```diff
@@ -857,16 +955,17 @@ catch {
...
                 Message = $errorMessage
                 IsError = $true
             })
     }
 }
 finally {
-    # Check if auditLogs contains errors, if errors are found, set success to false
+    # Check if auditLogs contains errors, if no errors are found, set success to true
     if ($outputContext.AuditLogs.IsError -contains $true) {
         $outputContext.Success = $false
```
Comment (Rick-Jongbloed) [inline update.ps1]: Success is already false

File: configuration.json [hunk 2]
```diff
@@ -74,7 +74,7 @@
     "type": "radio",
     "defaultValue": "true",
     "templateOptions": {
-      "label": "When an item can't be found in  TOPdesk",
+      "label": "When an item can't be found in  Topdesk",
```
[the remark this answers is not in the document]
Comment (Rick-Jongbloed) [reply configuration.json]: Fixed

File: delete.ps1 [hunk 3]
```diff
@@ -400,16 +501,17 @@ catch {
...
                 Message = $errorMessage
                 IsError = $true
             })
     }
 }
 finally {
-    # Check if auditLogs contains errors, if errors are found, set success to false
+    # Check if auditLogs contains errors, if no errors are found, set success to true
     if ($outputContext.AuditLogs.IsError -contains $true) {
         $outputContext.Success = $false
```
[the remark this answers is not in the document]
Comment (Rick-Jongbloed) [reply delete.ps1]: Fixed myself as discussed

File: configuration.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rick-Jongbloed) [reply configuration.json]: Fixed myself as discussed

File: update.ps1 (hunk 1 above)
Comment (Rick-Jongbloed) [reply update.ps1]: Fixed myself as discussed

File: permissions/incident/revokePermission.ps1 [hunk 4]
```diff
@@ -1049,8 +1042,11 @@ catch {
     # End
 }
 finally {
-    # Check if auditLogs contains errors, if errors are found, set succes to false
+    # Check if auditLogs contains errors, if no errors are found, set success to true
     if ($outputContext.AuditLogs.IsError -contains $true) {
         $outputContext.Success = $false
```
[the remark this answers is not in the document]
Comment (Rick-Jongbloed) [reply permissions/incident/revokePermission.ps1]: Fixed myself as discussed

File: permissions/incident/grantPermission.ps1 [hunk 5]
```diff
@@ -1049,8 +1042,11 @@ catch {
     # End
 }
 finally {
-    # Check if auditLogs contains errors, if errors are found, set succes to false
+    # Check if auditLogs contains errors, if no errors are found, set success to true
     if ($outputContext.AuditLogs.IsError -contains $true) {
         $outputContext.Success = $false
```
[the remark this answers is not in the document]
Comment (Rick-Jongbloed) [reply permissions/incident/grantPermission.ps1]: Fixed myself as discussed

File: permissions/change/revokePermission.ps1 [hunk 6]
```diff
@@ -898,8 +891,11 @@ catch {
     # End
 }
 finally {
-    # Check if auditLogs contains errors, if errors are found, set succes to false
+    # Check if auditLogs contains errors, if no errors are found, set success to true
     if ($outputContext.AuditLogs.IsError -contains $true) {
         $outputContext.Success = $false
```
[the remark this answers is not in the document]
Comment (Rick-Jongbloed) [reply permissions/change/revokePermission.ps1]: Fixed myself as discussed

File: permissions/change/grantPermission.ps1 [hunk 7]
```diff
@@ -898,8 +891,11 @@ catch {
     # End
 }
 finally {
-    # Check if auditLogs contains errors, if errors are found, set succes to false
+    # Check if auditLogs contains errors, if no errors are found, set success to true
     if ($outputContext.AuditLogs.IsError -contains $true) {
         $outputContext.Success = $false
```
[the remark this answers is not in the document]
Comment (Rick-Jongbloed) [reply permissions/change/grantPermission.ps1]: Fixed myself as discussed

File: enable.ps1 [hunk 8]
```diff
@@ -351,9 +334,11 @@ catch {
     }
 }
 finally {
-    # Check if auditLogs contains errors, if errors are found, set success to false
+    # Check if auditLogs contains errors, if no errors are found, set success to true
     if ($outputContext.AuditLogs.IsError -contains $true) {
         $outputContext.Success = $false
```
[the remark this answers is not in the document]
Comment (Rick-Jongbloed) [reply enable.ps1]: Fixed myself as discussed

File: disable.ps1 [hunk 9]
```diff
@@ -340,16 +327,17 @@ catch {
...
                 Message = $errorMessage
                 IsError = $true
             })
     }
 }
 finally {
-    # Check if auditLogs contains errors, if errors are found, set success to false
+    # Check if auditLogs contains errors, if no errors are found, set success to true
     if ($outputContext.AuditLogs.IsError -contains $true) {
         $outputContext.Success = $false
```
[the remark this answers is not in the document]
Comment (Rick-Jongbloed) [reply disable.ps1]: Fixed myself as discussed

**Review by Rick-Jongbloed — APPROVED** [review]
All requested changes are implemented. Code looks good. Approved.

**Review by Rick-Jongbloed — APPROVED** [review]

**Review by Rick-Jongbloed — APPROVED** [review]

**Pull request merged**